IdentityCore/src/webview/embeddedWebview/challangeHandlers/mac/MSIDCertAuthHandler.m

```diff
@@ -72,7 +76,16 @@ + (BOOL)handleChallenge:(NSURLAuthenticationChallenge *)challenge
                  completionHandler(NSURLSessionAuthChallengeRejectProtectionSpace, nil);
                  return;
              }
-             
+            
+            // If there is no preferred identity saved, we must set preferred identity certificate using hostname and key usage parameters
+            if (identity == NULL)
+            {
+                NSArray *arr = @[(__bridge NSString *)kSecAttrCanSign, (__bridge NSString *)kSecAttrCanEncrypt, (__bridge NSString *)kSecAttrCanDecrypt];
+                CFArrayRef arrayRef = (__bridge CFArrayRef)arr;
+                OSStatus status = SecIdentitySetPreferred(selectedIdentity, (CFStringRef)host, arrayRef);
+                MSID_LOG_WITH_CTX(MSIDLogLevelInfo, context, @"Result of setting identity preference is %d", status);
```

Comment:
It is more probable that this operation is successful and the logs would be littered with status = 0. Lets log here with error message only when status != 0